### PR TITLE
test: 00274-extreme-integers-comparator test inputs of different lengths, large numbers

### PR DESCRIPTION
**Pull request by MajorLift**
1. The existing test cases all compare integers of equal lengths. 
2. Large inputs aren't covered. Added test cases for `Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER`.

See: #21601

### CODE REVIEW

**Comment by Alexsey** [comment]
Having integers of different lengths is a good idea

Adding big integers would increase the problem's complexity because solutions that work with small integers can be much simpler than for bit ones. For example,
* [solution with limit](https://github.com/type-challenges/type-challenges/issues/11435)
* [solution with no limit](https://github.com/type-challenges/type-challenges/issues/11444)

So it depends on whether we want to make the problem harder

**Comment by MajorLift** [comment]
Agreed. I'll leave it up to @antfu to cherry-pick or just merge. 

In general, I think it might be worth adding a commented-out "extensions" section to the repo's template for test cases that the user can opt-in to when attempting advanced solutions that are more complex or require handling more edge cases.

File: questions/00274-extreme-integers-comparator/test-cases.ts [hunk 1]
```diff
@@ -18,4 +18,18 @@ type cases = [
...
   Expect<Equal<Comparator<-38, -38>, Comparison.Equal>>,
+
+  Expect<Equal<Comparator<1, 100>, Comparison.Lower>>,
+  Expect<Equal<Comparator<100, 1>, Comparison.Greater>>,
+  Expect<Equal<Comparator<-100, 1>, Comparison.Lower>>,
+  Expect<Equal<Comparator<1, -100>, Comparison.Greater>>,
+  Expect<Equal<Comparator<-100, -1>, Comparison.Lower>>,
+  Expect<Equal<Comparator<-1, -100>, Comparison.Greater>>,
+
+  Expect<Equal<Comparator<9007199254740992, 9007199254740992>, Comparison.Equal>>,
```
Comment (antfu) [inline questions/00274-extreme-integers-comparator/test-cases.ts:29]:
```suggestion
  // Extra tests if you like to challenge yourself!
  Expect<Equal<Comparator<9007199254740992, 9007199254740992>, Comparison.Equal>>,
```
